consul/datadog_checks/consul/consul.py

```diff
@@ -103,11 +105,19 @@ def consul_request(self, instance, endpoint):
             else:
                 resp = requests.get(url, verify=cabundlefile, headers=headers)
 
+            resp.raise_for_status()
+
         except requests.exceptions.Timeout:
             self.log.exception('Consul request to {} timed out'.format(url))
+            self.service_check(self.CONSUL_CAN_CONNECT, self.CRITICAL, tags=service_check_tags)
+            raise
+        except Exception:
+            self.log.exception("Consul request to {} failed".format(url))
+            self.service_check(self.CONSUL_CAN_CONNECT, self.CRITICAL, tags=service_check_tags)
             raise
```

Comment:
Since the exceptions get raised for `consul_request`, we'd still only send a CRITICAL service check for the first endpoint that fails right? (Just confirming this is the case so that if all endpoints are down at the same time, there wouldn't be a flood of alerts)

Reply:
Yeah the only place where the exception is caught is when sending the `consul.up` service check. So if all the endpoints fail, only one service check would be sent, and I think the proper way to monitor it would be to create a cluster monitor across the different endpoints to trigger if one of the checks fail. Does that make sense ?

Reply:
Yeah I think that makes sense and would avoid a flood of alerts for the same problem.